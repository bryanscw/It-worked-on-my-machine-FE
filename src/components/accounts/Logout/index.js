import React, { Component } from 'react';

export class Logout extends Component {
    render() {
        return (
            <div className="container">
                This is a logout page
            </div>
        )
    }
}

export default Logout
