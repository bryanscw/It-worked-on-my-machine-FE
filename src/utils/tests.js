import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    return <Provider store={store}>{component}</Provider>;
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    return <Router history={history}>{component}</Router>
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    return renderWithRedux(renderWithRouter(component, route), initialState);
}