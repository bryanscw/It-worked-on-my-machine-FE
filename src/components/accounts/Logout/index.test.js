import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Logout from './index.js';
import { renderWithReduxRouter } from '../../../utils/tests.js';

afterEach(cleanup);

it('should take a snapshot', () => {
    const { asFragment } = renderWithReduxRouter(<Logout />);
    expect(asFragment()).toMatchSnapshot();
})
