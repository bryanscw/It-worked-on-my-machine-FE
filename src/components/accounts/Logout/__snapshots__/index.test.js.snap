// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      You have logged out
    </h1>
    <p>
      Thanks for visiting.
    </p>
    <a
      class="btn btn-primary"
      href="/login"
    >
      Login
    </a>
  </div>
</DocumentFragment>
`;
